python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -165,6 +159,12 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
         .timestamp_type(cpp_timestamp_type)
         .build()
     )
+    cdef vector[string] cpp_columns
+    if columns is not None:
+        cpp_columns.reserve(len(columns))
+        for col in columns or []:
+            cpp_columns.push_back(str(col).encode())
+        args.set_columns(cpp_columns)
 
```

Comment:
and then after line 220, inside `if is_range_index:` block, you need to add this condition:
```python
if not allow_index:
    return df
```

It would look like:
```python
# Set the index column
if index_col is not None and len(index_col) > 0:
    if is_range_index:
        if not allow_index:
            return df
        range_index_meta = index_col[0]
```